pkg/spec/spec.go

```diff
@@ -533,11 +534,27 @@ func addRlimits(config *CreateConfig, g *generate.Generator) error {
 	// If not explicitly overridden by the user, default number of open
 	// files and number of processes to the maximum they can be set to
 	// (without overriding a sysctl)
-	if !nofileSet && !isRootless {
-		g.AddProcessRlimits("RLIMIT_NOFILE", kernelMax, kernelMax)
+	if !nofileSet {
+		if isRootless {
+			var rlimit unix.Rlimit
+			if err := unix.Getrlimit(unix.RLIMIT_NOFILE, &rlimit); err != nil {
+				return errors.Wrapf(err, "failed to return RLIMIT_NOFILE ulimit")
+			}
+			g.AddProcessRlimits("RLIMIT_NOFILE", rlimit.Cur, rlimit.Max)
+		} else {
+			g.AddProcessRlimits("RLIMIT_NOFILE", kernelMax, kernelMax)
```

Comment:
Why is a hard-coded constant of `1048576` used for `kernelMax` for both number of files and number of processes?  Is there a proper default that could be determined from the environment?

Reply:
Don't know, I was trying to fix rootless case.  I would actually think we should just use the current environment in both cases.  But not sure.  I am slightly worried about root case being able to override default MAX if the process has CAP_SYS_RESOURCE.  
```
man capabilities
...
       CAP_SYS_RESOURCE
              * Use reserved space on ext2 filesystems;
              * make ioctl(2) calls controlling ext3 journaling;
              * override disk quota limits;
              * increase resource limits (see setrlimit(2));
              * override RLIMIT_NPROC resource limit;
              * override maximum number of consoles on console allocation;
              * override maximum number of keymaps;
              * allow more than 64hz interrupts from the real-time clock;
              * raise  msg_qbytes  limit  for  a  System  V  message  queue  above  the  limit  in
                /proc/sys/kernel/msgmnb (see msgop(2) and msgctl(2));
              * allow  the RLIMIT_NOFILE resource limit on the number of "in-flight" file descrip‐
                tors to be bypassed when passing file descriptors to another process  via  a  UNIX
                domain socket (see unix(7));
              * override  the /proc/sys/fs/pipe-size-max limit when setting the capacity of a pipe
                using the F_SETPIPE_SZ fcntl(2) command.
              * use F_SETPIPE_SZ to increase the capacity of a pipe above the limit  specified  by
                /proc/sys/fs/pipe-max-size;
              * override  /proc/sys/fs/mqueue/queues_max  limit when creating POSIX message queues
                (see mq_overview(7));
              * employ the prctl(2) PR_SET_MM operation;
              * set /proc/[pid]/oom_score_adj to a value lower  than  the  value  last  set  by  a
                process with CAP_SYS_RESOURCE.
```

@giuseppe @mheon Any idea where this kernelMax came from?

Reply:
I think it is just the default used by Docker.

https://github.com/containers/buildah/commit/a2b018430df1e8e89b23cb5bfa49e4d3517e1c2d


Also, shouldn't these changes be done in `pkg/specgen` instead of `pkg/spec`?

Reply:
It is being done in both places.  pkg/spec is still used by varlink.